src/sql/workbench/contrib/notebook/browser/markdownToolbarActions.ts

```diff
@@ -179,14 +161,32 @@ export class MarkdownTextTransformer {
 
 			let editorModel = editorControl.getModel() as TextModel;
 			let isUndo = false;
-			if (editorModel) {
-				let markdownLineType = getMarkdownLineType(type);
-				// Paragraph (empty string) is just used for replacing any existing headers so will never be an undo operation
-				isUndo = beginInsertedText && this.isUndoOperation(selection, type, markdownLineType, editorModel);
-				if (isUndo) {
-					this.handleUndoOperation(markdownLineType, startRange, endRange, editorModel, beginInsertedText, endInsertedText, selections, selection);
-				} else {
-					this.handleTransformOperation(markdownLineType, type, startRange, endRange, editorModel, beginInsertedText, endInsertedText, selections, selection);
+
+			// If markdown has been previously transformed (e.g. through a callout), insert it "as is"
```

Comment:
I'm a little confused why you're using something called `MarkdownTextTransformer` if you're not even transforming the text. All you're doing is just inserting the text right? Seems like that could just be done in a separate standalone function or something - no reason to overload the meaning of this class.

Reply:
> Tested linking and looks good! Theres still that issue of ESC key closing the dialog and exiting edit mode though.

Yeah that'll be tackled separately in a different PR.

Reply:
Thanks. The logic is now in its own standalone function.